DEVELOPER_GUIDE.md

```diff
@@ -616,6 +616,7 @@ Pass a list of files or directories to limit your search.
 
 The Github workflow in [lucene-snapshots.yml](.github/workflows/lucene-snapshots.yml) is a Github worfklow executable by maintainers to build a top-down snapshot build of lucene.
```

Comment:
While you're here editing the developer guide, can we fix this typo in `workflow`?